lib/bone.js

```diff
@@ -757,14 +761,32 @@ class Bone {
   }
 
   static async bulkCreate(records) {
+    const { driver, table, attributes, primaryKey, primaryColumn } = this;
+    /**
+     * set defaultValue when create
+     * @param {object} values
+     * @memberof Bone [private]
+     */
+    const _setDefaultValue = (record) => {
+      for (const name in attributes) {
+        const value = record[name];
+        // set defaultValues
+        if (!value && !attributes[name].allowNull
+          && attributes[name].defaultValue !== undefined
+          && attributes[name].defaultValue !== null) {
+          record[name] = attributes[name].defaultValue;
+        }
```

Comment:
```suggestion
const { defaultValue }  = attributes[name];
if (value === undefined && defaultValue != null) {
  record[name] = defaultValue
}
```

value might be falsy values such as 0 or '', in which case the defaultValue should not take precedence. It is also possible that the value is null and defaultValue is something else (such as 0), in that case value = null should not be overwritten.

allowNull should not take part here. if defaultValue is declared, it should be used whenever possible.